sklearn/model_selection/_validation.py

```diff
@@ -128,16 +173,43 @@ def cross_val_score(estimator, X, y=None, groups=None, scoring=None, cv=None,
     X, y, groups = indexable(X, y, groups)
 
     cv = check_cv(cv, y, classifier=is_classifier(estimator))
-    scorer = check_scoring(estimator, scoring=scoring)
+    scorers, is_multimetric = _check_multimetric_scoring(estimator,
+                                                         scoring=scoring)
+    if not is_multimetric:
+        scorers = scorers['score']
+
     # We clone the estimator to make sure that all the folds are
     # independent, and that it is pickle-able.
     parallel = Parallel(n_jobs=n_jobs, verbose=verbose,
                         pre_dispatch=pre_dispatch)
-    scores = parallel(delayed(_fit_and_score)(clone(estimator), X, y, scorer,
-                                              train, test, verbose, None,
-                                              fit_params)
-                      for train, test in cv.split(X, y, groups))
-    return np.array(scores)[:, 0]
+    scores = parallel(delayed(_fit_and_score)(
+            clone(estimator), X, y, scorers, train, test, verbose, None,
+            fit_params, return_train_score=return_train_score,
+            # We need the times only for multimetric scoring
+            return_times=is_multimetric)
+        for train, test in cv.split(X, y, groups))
+
+    if not is_multimetric:
+        return np.array(next(zip(*scores)))
+
+    if return_train_score:
```

Comment:
should value error on single metric with a nice message. before computing anything.